pkg/compactor/shard_aware_deduplicate_filter.go

```diff
@@ -60,11 +61,10 @@ func (f *ShardAwareDeduplicateFilter) Filter(ctx context.Context, metas map[ulid
 	return nil
 }
 
-// findDuplicates finds blocks that are fully included in other blocks.
-// It works by 1) sorting blocks by number of sources, and 2) iterating through each block, and adding it to the
-// tree of blocks, where children are successors of given block.
+// findDuplicates finds all the blocks from the input slice of blocks that are fully included in other blocks from the
+// input slice. The found blocks are returned as a map whose keys are blocks' ULIDs.
```

Comment:
[Nit] I think this reads a bit better (also "whose" should refer to people):
```suggestion
// findDuplicates finds all the blocks from the input slice of blocks that are fully included in other blocks
// within the same slice. The found blocks are returned as a map which keys are blocks' ULIDs.
```